core/capabilities/ccip/ccipsolana/executecodec.go

```diff
@@ -0,0 +1,258 @@
+	}
+
+	chainReport := report.ChainReports[0]
+	if len(chainReport.Messages) > 1 {
+		return nil, fmt.Errorf("unexpected report message length: %d", len(chainReport.Messages))
+	}
+
+	var message ccip_router.Any2SVMRampMessage
+	var offChainTokenData [][]byte
+	if len(chainReport.Messages) > 0 {
+		// currently only allow executing one message at a time
+		msg := chainReport.Messages[0]
+		tokenAmounts := make([]ccip_router.Any2SVMTokenTransfer, 0, len(msg.TokenAmounts))
+		for _, tokenAmount := range msg.TokenAmounts {
+			if tokenAmount.Amount.IsEmpty() {
+				return nil, fmt.Errorf("empty amount for token: %s", tokenAmount.DestTokenAddress)
+			}
+
+			if len(tokenAmount.DestTokenAddress) != solana.PublicKeyLength {
+				return nil, fmt.Errorf("invalid destTokenAddress address: %v", tokenAmount.DestTokenAddress)
+			}
+
+			tokenAmounts = append(tokenAmounts, ccip_router.Any2SVMTokenTransfer{
+				SourcePoolAddress: tokenAmount.SourcePoolAddress,
+				DestTokenAddress:  solana.PublicKeyFromBytes(tokenAmount.DestTokenAddress),
+				ExtraData:         tokenAmount.ExtraData,
+				Amount:            ccip_router.CrossChainAmount{LeBytes: [32]uint8(encodeBigIntToFixedLengthLE(tokenAmount.Amount.Int, 32))},
+				DestGasAmount:     bytesToUint32LE(tokenAmount.DestExecData),
+			})
+		}
+
+		family, err := chainsel.GetSelectorFamily(uint64(chainReport.SourceChainSelector))
+		if err != nil {
+			return nil, fmt.Errorf("invalid source chain selector: %w", err)
+		}
+
+		// if source chain is Solana the Borsh decoder will be used, for EVM we will construct the extra args from
+		// the chain agnostic extra args map
+		var extraArgs ccip_router.SVMExtraArgs
+		switch family {
+		case chainsel.FamilySolana:
+			decoder := agbinary.NewBorshDecoder(msg.ExtraArgs)
+			err = extraArgs.UnmarshalWithDecoder(decoder)
+			if err != nil {
+				return nil, fmt.Errorf("invalid extra arguments: %w", err)
+			}
+		case chainsel.FamilyEVM:
+			extraArgs, err = parseExtraArgsMap(msg.ExtraArgsDecoded)
+			if err != nil {
+				return nil, fmt.Errorf("invalid extra args map: %w", err)
+			}
+		default:
+			return nil, fmt.Errorf("unsupported source chain family: %s", family)
+		}
+
+		if len(msg.Receiver) != solana.PublicKeyLength {
+			return nil, fmt.Errorf("invalid receiver address: %v", msg.Receiver)
+		}
+
+		message = ccip_router.Any2SVMRampMessage{
+			Header: ccip_router.RampMessageHeader{
+				MessageId:           msg.Header.MessageID,
+				SourceChainSelector: uint64(msg.Header.SourceChainSelector),
+				DestChainSelector:   uint64(msg.Header.DestChainSelector),
+				SequenceNumber:      uint64(msg.Header.SequenceNumber),
+				Nonce:               msg.Header.Nonce,
+			},
+			Sender:        msg.Sender,
+			Data:          msg.Data,
+			LogicReceiver: solana.PublicKeyFromBytes(msg.Receiver),
+			// TODO how to get the TokenReceiver ?
+			//TokenReceiver: solana.PublicKeyFromBytes(msg.Receiver),
```

Comment:
I believe it is this one: https://github.com/smartcontractkit/chainlink/blob/4e284976ea8ca7c0e355efce6336742d70918ac2/contracts/src/v0.8/ccip/libraries/Client.sol#L61

But we should confirm with EVM and Solana onchain team